data_service/config/logging_config.py

```diff
@@ -1,69 +1,115 @@
-SERVICE_NAME = "data-service"
-host = environment.get("DOCKER_HOST_NAME")
-command = json.dumps(sys.argv)
-
+request_start_time: ContextVar[int] = ContextVar("request_start_time")
+correlation_id: ContextVar[str] = ContextVar("correlation_id")
+method: ContextVar[str] = ContextVar("method")
+url: ContextVar[str] = ContextVar("url")
+remote_host: ContextVar[str] = ContextVar("remote_host")
+response_status: ContextVar[int] = ContextVar("response_status")
+response_time_ms: ContextVar[int] = ContextVar("response_time_ms")
+
+
+class RequestInfoFilter(logging.Filter):
+    def filter(self, record: logging.LogRecord) -> bool:
+        # Make sure string only contains alphanumeric characters, underscores and/or dashes
+        record.correlation_id = re.sub(r"[^\w\-]", "", correlation_id.get(""))
+        record.method = method.get("")
+        record.url = url.get("")
+        record.remote_host = remote_host.get("")
+        record.response_status = response_status.get("")
+        record.response_time_ms = response_time_ms.get("")
+        return True
+
+
+class MicrodataJSONFormatter(logging.Formatter):
+    def __init__(self):
+        self.pkg_meta = _get_project_meta()
+        self.host = environment.get("DOCKER_HOST_NAME")
+        self.command = json.dumps(sys.argv)
+
+    def format(self, record: logging.LogRecord) -> str:
+        return json.dumps(
+            {
+                "@timestamp": datetime.datetime.fromtimestamp(
+                    record.created,
+                    tz=datetime.timezone.utc,
+                ).isoformat(),
+                "command": self.command,
+                "error.stack": record.__dict__.get("exc_info"),
+                "host": self.host,
+                "message": record.getMessage(),
+                "level": record.levelno,
+                "levelName": record.levelname,
+                "loggerName": record.name,
+                "method": record.__dict__.get("method"),
+                "responseTime": record.__dict__.get("response_time_ms"),
+                "schemaVersion": "v3",
+                "serviceName": "data-service",
+                "serviceVersion": str(self.pkg_meta["version"]),
+                "source_host": record.__dict__.get("remote_host"),
+                "statusCode": record.__dict__.get("response_status"),
+                "thread": record.threadName,
+                "url": record.__dict__.get("url"),
+                "xRequestId": record.__dict__.get("correlation_id"),
+            }
+        )
 
-class CustomJSONLog(json_logging.JSONLogWebFormatter):
-    """
-    Customized application logger
-    """
 
-    def _format_log_object(self, record, request_util):
-        json_log_object = super(CustomJSONLog, self)._format_log_object(
-            record, request_util
+def setup_logging(app, log_level=logging.INFO):
+    @app.middleware("http")
+    async def add_process_time_header(request: Request, call_next: Callable):
+        logger = logging.getLogger()
```

Comment:
Looks like the logger is setup on every request. I think it should be outside the `@app.middleware` block.